### PR TITLE
Fix #919 - Support backword compatibility for configs

### LINKED ISSUE #919: Support standard introspection for oauth2 token validation
### Describe your problem(s)
MGW at the moment supports connecting with WSO2 APIM in order to validate opaque oauth2 tokens. It communicates to the key validation service of the WSO2 APIM.
We need to support validating standard oauth2 opaque token using the standard introspection endpoint.
### Describe your solution
Introduce a new security handler in order to validate with standard introspect endpoint.

### PR DESCRIPTION
**Pull request by Rajith90**
### Purpose
This will make the configuration backward compatible.
So ideally the version 3.0.x key manger config[1] will also be supported

[1] - 
```toml
[keyManager]
serverUrl="https://localhost:9443"
username="admin"
password="admin"
tokenContext="oauth2"
timestampSkew=5000
```

### Issues

Fixes #919 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 1]
```diff
@@ -123,11 +123,19 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
         },
         verifyHostname: getConfigBooleanValue(HTTP_CLIENTS_INSTANCE_ID, ENABLE_HOSTNAME_VERIFICATION, true)
     };
+
     http:OutboundAuthConfig? auth = ();
+    // support backward compatibility in reading the basic auth credentials when connecting with KM.
+    string username = getConfigValue(KM_CONF_INSTANCE_ID, USERNAME, "");
+    string password = getConfigValue(KM_CONF_INSTANCE_ID, PASSWORD, "");
+    if (username.length() == 0 && password.length() == 0) {
+        username = getConfigValue(KM_CONF_SECURITY_BASIC_INSTANCE_ID, USERNAME, "admin");
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: With amali's PR #948 for default configs, we have constants for default values of username and password. Let's use those after merging that PR.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 2]
```diff
@@ -120,11 +120,19 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
         },
         verifyHostname: getConfigBooleanValue(HTTP_CLIENTS_INSTANCE_ID, ENABLE_HOSTNAME_VERIFICATION, true)
     };
+
     http:OutboundAuthConfig? auth = ();
-    if (getConfigBooleanValue(KM_CONF_SECURITY_BASIC_INSTANCE_ID, ENABLED, DEFAULT_KM_CONF_SECURITY_BASIC_ENABLED)) {
+    // support backward compatibility in reading the basic auth credentials when connecting with KM.
+    string username = getConfigValue(KM_CONF_INSTANCE_ID, USERNAME, "");
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]:
```suggestion
    string username = getConfigValue(KM_CONF_INSTANCE_ID, USERNAME, DEFAULT_USERNAME);
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 3]
```diff
@@ -120,11 +120,19 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
...
+    string password = getConfigValue(KM_CONF_INSTANCE_ID, PASSWORD, "");
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]:
```suggestion
    string password = getConfigValue(KM_CONF_INSTANCE_ID, PASSWORD, DEFAULT_PASSWORD);
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal (hunk 2 above)
Comment (Rajith90) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: We can't have default values there. This is to support the old config. The empty string determines that old config is not used

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal (hunk 3 above)
Comment (Rajith90) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: We can't have default values there. This is to support the old config. The empty string determines that old config is not used

**Pull request merged**